Automattic/jetpack
# i18n-loader-webpack-plugin: Move the actual downloading out of the runtime

#### Changes proposed in this Pull Request:


Instead of being provided state that it uses to build the URL and fetch
it, i18n-loader-webpack-plugin now expects to be provided a loader that will do
all the work for it.
    
This makes the runtime smaller and allows for more flexibility in the future if
we need to change how the fetching works.

The Assets package provides the necessary loader module, as it did the state
module before. The state module is deprecated.

#### Jetpack product discussion


None.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Load or generate a language pack that provides translations for Instant Search's lazy module. Make sure its UI is still translated.
* For testing on wpcom, you'll have to grab automattic/jetpack-assets out of the build artifact and put it into `wp-content/mu-plugins/jetpack-packages/production/jetpack_vendor/`.
  * Also, before this can be deployed to wpcom, we'll similarly have to do a release of assets and update it on wpcom.
